lib/ansible/modules/cloud/dimensiondata/dimensiondata_firewall.py

```diff
@@ -0,0 +1,510 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016 Dimension Data
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#   - Aimon Bustardo <Aimon.Bustardo@dimensiondata.com>
+#   - Lawrence Lui   <lawrence.lui@dimensiondata.com>
+#   - Jay Riddell    <Jay.Riddell@dimensiondata.com>
+#   - Adam Friedman  <tintoy@tintoy.io>
+#
+
+DOCUMENTATION = '''
+---
+module: dimensiondata_firewall
+short_description: Create, update, and delete MCP 2.0 firewall rules.
+description:
+    - Create, update, and delete MCP 2.0 firewall rules.
+    - Requires MCP 2.0.
+version_added: "2.2"
```

Comment:
Sorry, I really should have caught this one. I'll make a note to check it when porting the other outstanding PRs.